pkg/relayer/miner/gen/gen_fixtures.go

```diff
@@ -0,0 +1,334 @@
+// NB: ensure this code is never included in any normal builds.
+//go:build ignore
+
+// NB: package MUST be `main` so that it can be run as a binary.
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"flag"
+	"fmt"
+	"hash"
+	"log"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/pokt-network/poktroll/pkg/observable"
+	"github.com/pokt-network/poktroll/pkg/observable/channel"
+	"github.com/pokt-network/poktroll/pkg/relayer"
+	"github.com/pokt-network/poktroll/pkg/relayer/miner"
+	"github.com/pokt-network/poktroll/pkg/relayer/protocol"
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+)
+
+const (
+	defaultDifficultyBits       = 16
+	defaultFixtureLimitPerGroup = 5
+	defaultRandLength           = 16
+	defaultOutPath              = "relay_fixtures_test.go"
+)
+
+var (
+	// flagDifficultyBitsThreshold is the number of leading zero bits that a
+	// randomized, serialized relay must have to be included in the
+	// `marshaledMinableRelaysHex` slice which is generated. It is also used as
+	// the maximum difficulty allowed for relays to be included in the
+	// `marshaledUnminableRelaysHex` slice.
+	flagDifficultyBitsThreshold int
+
+	// flagFixtureLimitPerGroup is the number of randomized, serialized relays that will be
+	// generated for each of `marshaledMinableRelaysHex` and
+	// `marshaledUnminableRelaysHex`.
+	flagFixtureLimitPerGroup int
+
+	// flagRandLength is the number of random bytes used for randomized values
+	// during fixture generation.
+	flagRandLength int
+
+	// flagOut is the path to the generated file.
+	flagOut string
+)
+
+// TODO_TECHDEBT: remove once marshaling using canonical codec.
+type marshalable interface {
+	Marshal() ([]byte, error)
+}
+
+func init() {
+	flag.IntVar(&flagDifficultyBitsThreshold, "difficulty-bits-threshold", defaultDifficultyBits, "the number of leading zero bits that a randomized, serialized relay must have to be included in the `marshaledMinableRelaysHex` slice which is generated. It is also used as the maximum difficulty allowed for relays to be included in the `marshaledUnminableRelaysHex` slice.")
+	flag.IntVar(&flagFixtureLimitPerGroup, "fixture-limit-per-group", defaultFixtureLimitPerGroup, "the number of randomized, serialized relays that will be generated for each of `marshaledMinableRelaysHex` and `marshaledUnminableRelaysHex`.")
+	flag.IntVar(&flagRandLength, "rand-length", defaultRandLength, "the number of random bytes used for randomized values during fixture generation.")
```

Comment:
I could imagine that we may want to generated more sets of relays with different properties for more tests in the future. I think providing that degree is freedom is simple to do using vanilla flags and without overcomplicating the design. Any other test which needs similar fixtures can add a //go:build directive to itself which is where it would then express it's requirements via flags. At the moment there's only one place where we're using is so I opted to leverage the flag defaults and not pass any values but I think this degree of freedom is useful going forward.